File: code/151.go
/*
 * @Author: lmio
 * @Date: 2023-04-03 10:17:40
 * @LastEditTime: 2023-04-07 14:33:22
 * @FilePath: /leetcode/code/151.go
 * @Description:151. 反转字符串中的单词
 */
package code

import (
	."leetcode/utils"
	"strings"
)

func ReverseWords(s string) string {
	res := make([]byte, 0, len(s))
	left := len(s)-1
	right := left
	for left >= 0 {
		for left >= 0 && s[left] == ' ' {
			left--
		}
		right = left
		for left >= 0 && s[left] != ' ' {
			left--
		}
		res = append(res, s[left+1:right+1]...)
		res = append(res, ' ')
	}
	return strings.Trim(ToString(res), " ")
}

File: code/151.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-07-02 22:48:02
LastEditors: lmio 2091319361@qq.com
Description: 151. 反转字符串中的单词
'''

def reverseWords(s: str) -> str:
    s = s.strip()
    res = []
    i = len(s)-1
    while i > 0:
        j = i
        while j > 0 and s[j] != ' ':
            j -= 1
        res.append(s[j+1:i+1])
        while j > 0 and s[j] == ' ':
            j -= 1
        i = j
    return ' '.join(res)